### ISSUE TITLE
Amount of the fee invoice

### ISSUE THREAD

**Issue by nepet** [ElementsProject/peerswap#110]
An initiator (__alice__) of a swap should always pay for the on-chain fees. On a swap-out the non-initiating party of the swap (__bob__)  is in charge of the on-chain part. That is why we have the __fee invoice__ as a part of the protocol (see https://github.com/ElementsProject/peerswap/blob/e563b002ed3f66fbce71f8e07037784dfa940716/docs/peer-protocol.md#the-swap_out_agreement-message).

Currently our implementation only addresses the estimated fee for the `opening-tx` and uses this estimate as the amount of the fee invoice. I see two problems here:
1. What about a refund of the on-chain funds, the `csv path`. Even if we can not know the fee that will be used for this tx in the future, we should at least try to estimate it in such a way that the non-initiating swap partner gets some compensation for this case.
2. We do not know the absolute size of the transaction without reserving the inputs. Reserving the inputs upfront would be annoying as we also have to take care that we release them again in the case that the fee invoice is not payed. Also, this way an evil peer could lock up parts of my wallet for the time that we wait for the fee invoice to be payed for.

What do we do about this? An easy first solution could be that we use the following formula for the fee invoice amount:
```
(estimated_fee_opening_tx + estimated_fee_claim_by_csv_tx)*x
```
with some float `x` that we would hard code in a first version but that could be a config and part of the swap negotiation in the future.

In the future we could call this the `premium` with a negotiated factor `x` ^^.

What do you think?

**Comment by nepet** [comment]
As for now: The fee invoice should compensate for the expected fees of the opening tx. This highly depends on the amount of inputs as well as on the type of input.

The general overhead of the transaction is:

| type | size [vbyte] |
| --- | --- |
| version | 4 |
| len in | 1 |
| len out | 1 |
| nLocktime | 4 |
| --- | --- |
| total | 10 |

The outputs of the opening tx have a size of:

| type | size [vbyte] |
| --- | --- |
| Return (P2WPKH) | 22 |
| Script (P2WSH) | 34 |
| Overhead (val(8), len(1)) | 2*9 |
| --- | --- |
| total | 74 

which totals to:
__overhead__ + __outputs__ = __84 vbyte__

If we expect _P2WPKH_ to be the most common input type we add _68 vByte_ for each input. This leads to the following linear expression:

size_P2WPKH = (68*n_inputs+84) vbyte

Given that the data on https://bitcoinvisuals.com/chain-input-count-tx is correct, the expected value of the number of inputs per transaction is 3.
This leads to a size expectation value of __288 vbyte__.

This is all under the assumption that the inputs are of _P2WPKH_. The size can easily double for different inputs.

My proposal here for the short term is:
We take the calculated avg of _288 vbyte_ with a security margin and use __350 vbyte__ for now until we have a better solution ready.

**Comment by nepet** [comment]
This is resolved by #126
